dawn/src/dawn/Optimizer/OptimizerContext.cpp

```diff
@@ -387,6 +388,16 @@ class StencilDescStatementMapper : public iir::ASTVisitor {
                        : LoopOrderKind::Backward);
     std::unique_ptr<Stage> stage =
         std::make_unique<Stage>(metadata_, instantiation_->nextUID(), interval);
+    // set the iteration space
+    // auto is = stmt->getVerticalRegion()->iterationSpace_;
+    int size = stmt->getVerticalRegion()->iterationSpace_.size();
+    std::array<std::optional<iir::Interval>, 2> intervals;
+    for(int i = 0; i < size; ++i) {
+      if(stmt->getVerticalRegion()->iterationSpace_[i]) {
+        intervals[i] = stmt->getVerticalRegion()->iterationSpace_[i].value();
+      }
+    }
+    stage->setIterationSpace(intervals);
```

Comment:
Add a constructor for Stage that accepts an iteration space.

Reply:
done